src/main/java/com/atlassian/bitbucket/jenkins/internal/status/BitbucketBuildStatusConstructor.java

```diff
@@ -0,0 +1,33 @@
+package com.atlassian.bitbucket.jenkins.internal.status;
+
+import com.atlassian.bitbucket.jenkins.internal.model.BitbucketBuildStatus;
+import com.atlassian.bitbucket.jenkins.internal.model.BuildState;
+import hudson.model.AbstractBuild;
+import hudson.model.Result;
+import org.jenkinsci.plugins.displayurlapi.DisplayURLProvider;
+
+import java.util.Arrays;
+import java.util.Collection;
+
+public class BitbucketBuildStatusConstructor {
```

Comment:
```suggestion
public class BitbucketBuildStatusFactory {
```

I feel `Constructor` is a reserved word, even if it is not technically I want to avoid confusion, and `Factory` works just as well